### PR TITLE
Unit tests for part of applyEvents

### PR DESCRIPTION
**Pull request by bowenxia**
**What changed?**
Added unit tests for part of applyEvents (one branch)
Lots of duplicated code like passing functions with a lot of parameters

**Why?**
code coverage week

**How did you test it?**
unit test

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: service/history/ndc/history_replicator_test.go [hunk 1]
```diff
@@ -380,3 +382,1110 @@ func TestApplyEvents(t *testing.T) {
...
+				mockReplicationTask.EXPECT().getDomainID().Return("test-domain-id").Times(1)
+				mockReplicationTask.EXPECT().getExecution().Return(&types.WorkflowExecution{
+					WorkflowID: "test-workflow-id",
+					RunID:      "test-run-id",
+				}).Times(1)
+				mockReplicationTask.EXPECT().getFirstEvent().Return(&types.HistoryEvent{
+					EventType: &workflowExecutionStartedType,
+				}).Times(1)
+			},
+			applyStartEventsFnAffordance: func() func(
```
Comment (Shaddoll) [inline service/history/ndc/history_replicator_test.go:422]: nit: you can just assign the function value returned by the func to the field, which makes the code easier to understand

File: service/history/ndc/history_replicator_test.go [hunk 2]
```diff
@@ -380,3 +382,1110 @@ func TestApplyEvents(t *testing.T) {
...
+					task replicationTask,
+					r *historyReplicatorImpl,
+				) error {
+					return nil
+				}
+				return fn
+			},
+			expectError: fmt.Errorf("test error"),
+		},
+		"Case3: case nil with !doContinue": {
```
Comment (Shaddoll) [inline service/history/ndc/history_replicator_test.go:780]: can you make the message more clear?

File: service/history/ndc/history_replicator_test.go (hunk 1 above)
Comment (davidporter-id-au) [reply service/history/ndc/history_replicator_test.go:422]: To nit: this is a mock you're assigning more than an affordance I think? it's a thing that you're expecting to get called as part of the test run?

File: service/history/ndc/history_replicator_test.go [hunk 3]
```diff
@@ -380,3 +382,1110 @@ func TestApplyEvents(t *testing.T) {
...
+			) (execution.MutableState, bool, error) {
+				fn := func(ctx ctx.Context,
+					context execution.Context,
+					mutableState execution.MutableState,
+					branchIndex int,
+					task replicationTask,
+					newConflictResolver newConflictResolverFn,
+				) (execution.MutableState, bool, error) {
+					return nil, false, fmt.Errorf("test error")
+				}
```
Comment (davidporter-id-au) [inline service/history/ndc/history_replicator_test.go:727]: You probably want to do at least some assertions here to ensure that you're passing in the right data: 

Most of the args are pretty hard to mess up, but Ints, strings, ints etc are much easier. Consider at least putting a couple of assertions here:
```
				fn := func(ctx ctx.Context,
					context execution.Context,
					mutableState execution.MutableState,
					branchIndex int,
					task replicationTask,
					newConflictResolver newConflictResolverFn,
				) (execution.MutableState, bool, error) {
                    assert.Equal(t, 123, branchIndex)
					return nil, false, fmt.Errorf("test error")
				}
				return fn
```

Comment (bowenxia) [reply service/history/ndc/history_replicator_test.go:584]: resolved offline

File: service/history/ndc/history_replicator_test.go [hunk 4]
```diff
@@ -23,6 +23,8 @@
 package ndc
 
 import (
+	ctx "context"
```
Comment (davidporter-id-au) [inline service/history/ndc/history_replicator_test.go:26]: nit: I'd avoid aliasing this just because it makes the package look like an instance variable

**Review by davidporter-id-au — APPROVED** [review]

**Review by Shaddoll — APPROVED** [review]

**Pull request merged**